crates/storage/provider/src/db_provider/block.rs

```diff
@@ -26,6 +26,15 @@ impl<DB: Database> HeaderProvider for ProviderImpl<DB> {
     }
 }
 
+impl<DB: Database> BlockHashProvider for ProviderImpl<DB> {
+    fn block_hash(&self, number: U256) -> Result<Option<H256>> {
+        // TODO: This unwrap is potentially unsafe
```

Comment:
When? Should this accept a u64? @rakita

Reply:
there should be something like and not use unwrap.
```
if let Some(number) = number.try_into() {

}
```
Probably it should use u64